middleware/proxy_headers.go

```diff
@@ -0,0 +1,50 @@
+package middleware
+
+import (
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/labstack/echo/v4"
+)
+
+var (
+	protoRegex = regexp.MustCompile(`(?i)(?:proto=)(https|http)`)
+)
+
+func ProxyHeaders() echo.MiddlewareFunc {
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) error {
+			if fwd := c.RealIP(); fwd != "" {
+				c.Request().RemoteAddr = fwd
+			}
+
+			if scheme := getScheme(c); scheme != "" {
+				c.Request().URL.Scheme = scheme
+			}
+
+			if c.Request().Header.Get(echo.HeaderXForwardedHost) != "" {
+				c.Request().Host = c.Request().Header.Get(echo.HeaderXForwardedHost)
```

Comment:
Getting host from proxied is little bit more complex. Please see https://github.com/labstack/echo/blob/master/ip.go